uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Import/Helpers/DataImportTrait.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+use CRM_HRLeaveAndAbsences_BAO_LeaveBalanceChange as LeaveBalanceChange;
+use CRM_HRLeaveAndAbsences_Service_LeaveRequestComment as LeaveRequestCommentService;
+use CRM_HRLeaveAndAbsences_BAO_AbsenceType as AbsenceType;
+
+trait CRM_HRLeaveAndAbsences_Import_Helpers_DataImportTrait {
+
+  /**
+   * @var array
+   *   Stores the type_id Balance change Option values.
+   */
+  private $balanceChangeTypes = [];
+
+  /**
+   * @var array
+   *   Stores the date_type Leave request Option values.
+   */
+  private $dayTypes = [];
+
+  /**
+   * @var LeaveRequestCommentService
+   */
+  private $leaveRequestCommentService;
+
+  /**
+   * @var array
+   */
+  private $sicknessAbsenceTypes;
+
+  /**
+   * Returns the type_id Balance change Option values.
+   *
+   * @return array
+   */
+  private function getBalanceChangeTypes() {
+    if (empty($this->balanceChangeTypes)) {
+      $this->balanceChangeTypes = array_flip(LeaveBalanceChange::buildOptions('type_id', 'validate'));
+    }
+
+    return $this->balanceChangeTypes;
+  }
+
+  /**
+   * Returns the date_type Leave request Option values.
+   *
+   * @return array
+   */
+  private function getDateTypes() {
+    if (empty($this->dayTypes)) {
+      $this->dayTypes = array_flip(LeaveRequest::buildOptions('from_date_type', 'validate'));
+    }
+
+    return $this->dayTypes;
+  }
+
+  /**
+   * Returns the date_type Leave request Option values.
+   *
+   * @return LeaveRequestCommentService
+   */
+  private function getLeaveRequestCommentService() {
+    if (empty($this->leaveRequestCommentService)) {
+      $this->leaveRequestCommentService = new CRM_HRLeaveAndAbsences_Service_LeaveRequestComment();
+    }
+
+    return $this->leaveRequestCommentService;
+  }
+
+  /**
+   * Returns Absence Types with is_sick as TRUE.
+   *
+   * @return array
+   */
+  private function getSicknessAbsenceTypes() {
+    if (empty($this->sicknessAbsenceTypes)) {
+      $absenceType = new AbsenceType();
+      $absenceType->is_active = 1;
+      $absenceType->find();
+
+      $sicknessAbsenceTypes = [];
+      while($absenceType->fetch()) {
+        if ($absenceType->is_sick) {
+          $sicknessAbsenceTypes[$absenceType->title] = $absenceType->id;
+        }
+      }
+      $this->sicknessAbsenceTypes = $sicknessAbsenceTypes;
```

Comment:
What about adding a new method in the AbsenceType BAO that returns the is_sick absence types?